<?xml version="1.0"?>
<launch>
    <arg name="robot_name" default="my_gen3"/>

    <!-- PID control: -->
    <group ns="/$(arg robot_name)">->

      <!-- Joint 1 control: -->
      <node name="controller" pkg="pid" type="controller" ns="joint_1" output="screen" >
        <param name="Kp" value="3.5" />
        <param name="Ki" value="0.0" />
        <param name="Kd" value="0.0" />
        <param name="max_loop_frequency" value="1000.0" />
        <param name="min_loop_frequency" value="800.0" />
      </node>

      <!-- Joint 2 control: -->
      <node name="controller" pkg="pid" type="controller" ns="joint_2" output="screen" >
        <param name="Kp" value="4.0" />
        <param name="Ki" value="0.0" />
        <param name="Kd" value="0.0" />
        <param name="max_loop_frequency" value="1000.0" />
        <param name="min_loop_frequency" value="800.0" />
      </node>

      <!-- Joint 3 control: -->
      <node name="controller" pkg="pid" type="controller" ns="joint_3" output="screen" >
        <param name="Kp" value="3.5" />
        <param name="Ki" value="0.0" />
        <param name="Kd" value="0.0" />
        <param name="max_loop_frequency" value="1000.0" />
        <param name="min_loop_frequency" value="800.0" />
      </node>

      <!-- Joint 4 control: -->
      <node name="controller" pkg="pid" type="controller" ns="joint_4" output="screen" >
        <param name="Kp" value="4.5" />
        <param name="Ki" value="0.0" />
        <param name="Kd" value="0.0" />
        <param name="max_loop_frequency" value="1000.0" />
        <param name="min_loop_frequency" value="800.0" />
      </node>

      <!-- Joint 5 control: -->
      <node name="controller" pkg="pid" type="controller" ns="joint_5" output="screen" >
        <param name="Kp" value="3.5" />
        <param name="Ki" value="0.0" />
        <param name="Kd" value="0.0" />
        <param name="max_loop_frequency" value="1000.0" />
        <param name="min_loop_frequency" value="800.0" />
      </node>

      <!-- Joint 6 control: -->
      <node name="controller" pkg="pid" type="controller" ns="joint_6" output="screen" >
        <param name="Kp" value="5.0" />
        <param name="Ki" value="0.0" />
        <param name="Kd" value="0.0" />
        <param name="max_loop_frequency" value="1000.0" />
        <param name="min_loop_frequency" value="800.0" />
      </node>

      <!-- Joint 7 control: -->
      <node name="controller" pkg="pid" type="controller" ns="joint_7" output="screen" >
        <param name="Kp" value="4.5" />
        <param name="Ki" value="0.0" />
        <param name="Kd" value="0.0" />
        <param name="max_loop_frequency" value="1000.0" />
        <param name="min_loop_frequency" value="800.0" />
      </node>
    </group>
</launch>